pkg/executor/mem_reader.go

```diff
@@ -726,11 +728,62 @@ func buildMemIndexLookUpReader(ctx context.Context, us *UnionScanExec, idxLookUp
 }
 
 func (m *memIndexLookUpReader) getMemRowsIter(ctx context.Context) (memRowsIter, error) {
-	data, err := m.getMemRows(ctx)
-	if err != nil {
-		return nil, errors.Trace(err)
+	kvRanges := [][]kv.KeyRange{m.idxReader.kvRanges}
```

Comment:
No new code are added.
This is indeed copied from `memRowsIterForTable` function

https://github.com/pingcap/tidb/pull/55922/files#diff-4745e9eb77a8450aa7d598691e57074ad2763cbb5570ad12493adba31bfb9310L955-L1010

Reply:
Do u mean copied from https://github.com/pingcap/tidb/blob/f9c4773ba34854cec93dec3c5cae8a625925f9e9/pkg/executor/mem_reader.go#L736-L794 ?

Reply:
Oh, yes!